pkg/driver/controller.go

```
@@ -749,3 +750,75 @@ func validateExistingAccessPoint(existingAccessPoint *cloud.AccessPoint, basePat
 
 	return nil
 }
+
+func findFileSystemId(ctx context.Context, volumeParams map[string]string) (string, error) {
+	if value, ok := volumeParams[FsId]; ok {
+		value = strings.TrimSpace(value)
+		if value != "" {
+			return value, nil
+		}
+	}
+
+	if ref, ok := volumeParams[FileSystemIdConfigRef]; ok {
+		return getFileSystemIdFromRef(ctx, "configmap", ref)
+	}
+
+	if ref, ok := volumeParams[FileSystemIdSecretRef]; ok {
+		return getFileSystemIdFromRef(ctx, "secret", ref)
+	}
+	return "", fmt.Errorf("fileSystemId not specified in volume parameters")
+}
+
+func getFileSystemIdFromRef(ctx context.Context, resourceType, ref string) (string, error) {
+	namespace, name, key := parseRef(ref)
+	if namespace == "" || name == "" || key == "" {
+		return "", fmt.Errorf("invalid %s reference %s/%s", resourceType, namespace, name)
+	}
+
+	client, err := cloud.DefaultKubernetesAPIClient()
+	if err != nil {
+		return "", err
+	}
+
+	var fsId string
+	if resourceType == "configmap" {
+		configMap, err := client.CoreV1().ConfigMaps(namespace).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return "", formatK8Errors(err, resourceType, namespace, name)
+		}
+		fsId = configMap.Data[key]
+	} else if resourceType == "secret" {
+		secret, err := client.CoreV1().Secrets(namespace).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return "", formatK8Errors(err, resourceType, namespace, name)
+		}
+		fsId = string(secret.Data[key])
+	} else {
+		return "", fmt.Errorf("unknown resource type %s", resourceType)
+	}
+
+	fsId = strings.TrimSpace(fsId)
+	if fsId == "" {
+		return "", fmt.Errorf("key %s not found in %s %s/%s", key, resourceType, namespace, name)
+	}
+
+	return fsId, nil
+}
+
+func parseRef(ref string) (string, string, string) {
+	ref = strings.TrimSpace(ref)
+	parts := strings.Split(ref, "/")
+	if len(parts) != 3 {
+		return "", "", ""
```

We should probably return error and propagate to other callers rather than returning empty strings here if failing to parse.